Src/AwesomeAssertions/Execution/LateBoundTestFramework.cs

```diff
@@ -23,20 +30,37 @@ public bool IsAvailable
     {
         get
         {
-            var assembly = FindExceptionAssembly();
-            var exceptionType = assembly?.GetType(ExceptionFullName);
+            Assembly? assembly = FindExceptionAssembly();
+            Type? exceptionType = assembly?.GetType(ExceptionFullName);
 
-            exceptionFactory = exceptionType != null
-                ? message => (Exception)Activator.CreateInstance(exceptionType, message)
+            exceptionFactory = exceptionType is not null
+                ? message => (Exception)Activator.CreateInstance(exceptionType, message)!
                 : _ => throw new InvalidOperationException($"{GetType().Name} is not available");
 
             return exceptionType is not null;
         }
     }
 
-    private Assembly FindExceptionAssembly()
+    protected internal abstract string AssemblyName { get; }
+
+    protected abstract string ExceptionFullName { get; }
+
+    private static IEnumerable<Assembly> GetAssemblies()
     {
-        var assembly = Array.Find(AppDomain.CurrentDomain.GetAssemblies(), a => a.GetName().Name == AssemblyName);
+#if NET6_0_OR_GREATER
+        // In some constellations a test framework assembly might have been loaded more than once. Make sure we get the correct one.
```

Comment:
"Make sure we get the correct one" does not really help to understand, what makes the difference here.
Could you please add some information what is different when using `AssemblyLoadContext` instead of `AppDomain`?